### ISSUE TITLE
Installer 18.7 doese freeze on Hyper-V

### ISSUE THREAD

**Issue by Schroeffu** [opnsense/core#2961]
The installer does freeze at the step "Select Task: Guided installation, Manual installation, Import configuration, Reset password, Reboot, Exit". 

Version: OPNsense-18.7-OpenSSL-dvd-amd64.iso
Happens on: Hyper-V

Screenshot: https://imgur.com/a/rSNq4qz

I have already tried the idea from https://forum.opnsense.org/index.php?topic=4470.msg17105#msg17105 to add set kern.vty="vt" and set console="efi" while booting - still freezing.  I also tried set kern.vty=sc but the its not booting anymore with exactly the same issue like here screenshoted: https://github.com/opnsense/core/issues/2254#issuecomment-382223999

I'am opening an issue here, because i really want to compare pfsense <> opnsense. pfsense was install-able on the very same hyper-v host without any issue, so, something is different to opnsense causing the issue. Sorry, there is no logs, its just freezing.

See also https://github.com/opnsense/core/issues/2254#issuecomment-372104272 , according to the comments there after closing the issue, other poeple are affected too.

**Comment by fichtner** [comment]
This is Gen2, right? We've been talking about this for a while. The frontend of the installer freezes, but the backend is still there. Try CTRL+C and log into the installer again and it should continue where it left off?

**Comment by Schroeffu** [comment]
Yes Gen2. You are right, entering CRTL+C and log-in again is workaround. At the point "Choose UEFI/MBR" mode it logs me out again. CRTL-C + relogin again was working. Great!

Now just my working colleagues got immediately IPs from OpenSense's DNS , aargh.. Had to stop the fresh install immediately again, but it was looking OK.

Anyway, I think with your fanstatic workaround its ok to install it on Hyper-V , thanks again!

**Comment by fichtner** [comment]
LAN and WAN interfaces might be reversed?

I'll look into the installer stability on Hyper-V for 19.1 further to hopefully fix it and bring it in line with how it works in all other HW and VM environments. :)

**Comment by Schroeffu** [comment]
> LAN and WAN interfaces might be reversed?
> 
> I'll look into the installer stability on Hyper-V for 19.1 further to hopefully fix it and bring it in line with how it works in all other HW and VM environments. :)

My fault, I meant DHCP (not DNS), OpnSense was giving immediately ips to other clients but it shouldn't, will check in an OpnSense Forum for help how to deactivate DHCP during installation of OpnSense.

Thanks for tracking the issue about this Hyper-V freeze.

**Comment by soder10** [comment]
@Schroeffu: try assigning a new Hyperv vSWITCH to your testing VM, that has no access to your prod. LAN, before rolling out any such trial VM into production. Any non-planned DHCP is dangerous to a prod network.

**Comment by fichtner** [comment]
Good news everyone! I found a way to reproduce a segfault while working on ZFS support for the installer... https://github.com/opnsense/bsdinstaller/commit/058fccb9f911

I can upload an image in a few days for the Hyper-V savvy to test it. :)

**Comment by fichtner** [comment]
A test image should be available later today... sorry for the delay.

**Comment by fichtner** [comment]
Here it is, please test :)

https://pkg.opnsense.org/FreeBSD:11:amd64/snapshots/OPNsense-201901021629-LibreSSL-dvd-amd64.iso.bz2

(image contains an intermediate state between 18.7.9 and 18.7.10 but should work fine in production including updates if required, but is not overly recommended)

**Comment by marjohn56** [comment]
Still the same, hangs ( or appears to ) at the Guided Install. CTRL C drops to prompt and restart installer then works,

**Comment by fichtner** [comment]
So the segfault is a different issue. Can you post /tmp/installer.log from when it hang (don't restart)? The hang is specific to Hyper-V for some reason... and when you "restart" does it stay in the crashed menu or does the navigation restart?

**Comment by fichtner** [comment]
another thing--- can you try the same with the early installer ?

When it prompts for "import" press enter, then type "!" and press enter. In the magic shell, type "opnsense-installer" and enter...

**Comment by marjohn56** [comment]
Early installer freezes at 'OK Lets Go' & I cannot break out of it.

In normal install I get to 'Accept these settings' and that's it, I can however then hit CTRL-C and break out. Log in again and away it will go and install. Now there's another thing I have discovered. If go for manual install then it gets as far as Edit Partitions, then freezes and it's eating CPU, so much that my cooler pump and fans go into overdrive. Again, CTRL-C will break out of it.

I've not configured the virtual switches yet on Hyper-v so cannot pull the log, I have however grabbed it and have some images for you. In order

![capture5](https://user-images.githubusercontent.com/17962918/50611955-36fc4680-0ed0-11e9-8b18-2afdc7b46162.PNG)
![capture1](https://user-images.githubusercontent.com/17962918/50611956-36fc4680-0ed0-11e9-91b8-73f85206b5c4.PNG)
![capture2](https://user-images.githubusercontent.com/17962918/50611957-36fc4680-0ed0-11e9-8156-f9b60f55ff37.PNG)
![capture3](https://user-images.githubusercontent.com/17962918/50611958-3794dd00-0ed0-11e9-9105-5a94978b518c.PNG)
![capture4](https://user-images.githubusercontent.com/17962918/50611959-3794dd00-0ed0-11e9-810b-07b66cde9d80.PNG)

**Comment by marjohn56** [comment]
Ok, had a few minutes to spare before zzzz.

Here's the installer log. I CTRL c'd and then logged in again and it worked, I'm sure you'll see that in the log.
[installer.log](https://github.com/opnsense/core/files/2722327/installer.log)

**Comment by marjohn56** [comment]
Here's one that I did not hit ctrl-c on. just left it and grabbed the log.
[installer_no_ctrl_c.log](https://github.com/opnsense/core/files/2722345/installer_no_ctrl_c.log)

**Comment by lhorace** [comment]
I might be able to lend a hand with this but the issue, the logs doesn't explain a lot for a passer by. The differences, is DFUI? Only people who are familiar can help :(

**Comment by lhorace** [comment]
For more context, I've come across a few people on the forums who are kvm based are experiencing 'random' refreezes. So far, I haven't changed CPU model on KVM side, I plan to test to using host-passthrough.

**Comment by fichtner** [comment]
Are we talking about installer "freezes" or system freezes? To me it seems the only way the installer freezes is when their local TCP traffic is dropped for some reason, hence the "pfctl -d" question.

This, however, has no association with freezes as the installer only appears to be frozen but the system (and backend) remains responsive. And this is only the case for Hyper-V.

**Comment by lhorace** [comment]
Ahh, right, forgot. Sorry.

**Comment by fichtner** [comment]
No worries.

**Comment by lhorace** [comment]
I watching a lot open source communities, I should have read up on this issue! I matched this problem with a freezes

**Comment by smoonlee** [comment]
This is still an issue in 19.1 iso?

**Comment by fichtner** [comment]
Yes.

Also when you log in as root, option 8, using "opnsense-installer" ?

**Comment by smoonlee** [comment]
> Yes.
> 
> Also when you log in as root, option 8, using "opnsense-installer" ?

Ok, Let me try this

**Comment by smoonlee** [comment]
Same issue, freezes on the Select Task Option and Ctrl + C and relaunch started the install process from the beginning.

**Comment by fichtner** [comment]
It would mean the backend dies a horrible death, is this in dmesg ? There's also /tmp/installer.log

**Comment by smoonlee** [comment]
That took longer than it should but heres the log from /tmp/installer.log

https://paste.pound-python.org/show/Uw6mZompIiyB1Quf3mXW/

**Comment by fichtner** [comment]
just stops after rendering the menu it seems. I'll try to replace the installer for 19.7, it was not built for modern consoles it seems... quite a bit of work unfortunately.

**Comment by Schroeffu** [comment]
Hi there, something has changed between 19.1 <> 18.7 installer? (OPNsense-19.1-OpenSSL-dvd-amd64.iso)

The workaround from 18.7 is not working anymore for me, `CRTL + C` while freezing installer process is also with 19.1 also killing the whole install-process, not only "minimize" the frozen installer.

So, the installer starts prompting from the beginning again after  CRTL+C & re-login as "installer/opnsense", and it freezes anytime again while asking for Swap Partition. 

So, could not install 19.1 on HyperV Gen2 so far :-/

**Comment by fichtner** [comment]
CTRL+C  should work again on 19.1.4 -- images will be available next week. Don't expect miracles just yet.
